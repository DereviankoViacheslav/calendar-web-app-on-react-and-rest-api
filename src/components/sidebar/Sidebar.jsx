import React from 'react';

function Sidebar() {
  const arrHoursElems = [...new Array(24)].map((el, index) => (
    <div key={index} className="hour">
      <span className="hour__text">{`${(index < 10) ? `0${index}` : index}:00`}</span>
    </div>));

  return (
    <div className="sidebar">
      {arrHoursElems}
    </div>
  );
}

export default Sidebar;
