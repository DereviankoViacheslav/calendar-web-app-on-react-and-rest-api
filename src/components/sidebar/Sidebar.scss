.sidebar {
  display: flex;
  flex-direction: column;
  color: #70757a;

  :first-child .hour__text {
    display: none;
  }
}

.hour {
  height: 42px;
  width: 55px;

  &__text {
    display: block;
    margin: -9px 0 0 10px;
  }
}