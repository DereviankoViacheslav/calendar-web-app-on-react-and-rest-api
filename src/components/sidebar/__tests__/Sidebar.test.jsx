import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from '../Sidebar'

describe('<Sidebar />', () => {
  it('Get snapshot', () => {
    const wrappedComponent = shallow(<Sidebar />);
    expect(wrappedComponent).toMatchSnapshot();
  });
});