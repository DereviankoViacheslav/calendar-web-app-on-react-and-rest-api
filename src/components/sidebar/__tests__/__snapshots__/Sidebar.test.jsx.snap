// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> Get snapshot 1`] = `
<div
  className="sidebar"
>
  <div
    className="hour"
    key="0"
  >
    <span
      className="hour__text"
    >
      00:00
    </span>
  </div>
  <div
    className="hour"
    key="1"
  >
    <span
      className="hour__text"
    >
      01:00
    </span>
  </div>
  <div
    className="hour"
    key="2"
  >
    <span
      className="hour__text"
    >
      02:00
    </span>
  </div>
  <div
    className="hour"
    key="3"
  >
    <span
      className="hour__text"
    >
      03:00
    </span>
  </div>
  <div
    className="hour"
    key="4"
  >
    <span
      className="hour__text"
    >
      04:00
    </span>
  </div>
  <div
    className="hour"
    key="5"
  >
    <span
      className="hour__text"
    >
      05:00
    </span>
  </div>
  <div
    className="hour"
    key="6"
  >
    <span
      className="hour__text"
    >
      06:00
    </span>
  </div>
  <div
    className="hour"
    key="7"
  >
    <span
      className="hour__text"
    >
      07:00
    </span>
  </div>
  <div
    className="hour"
    key="8"
  >
    <span
      className="hour__text"
    >
      08:00
    </span>
  </div>
  <div
    className="hour"
    key="9"
  >
    <span
      className="hour__text"
    >
      09:00
    </span>
  </div>
  <div
    className="hour"
    key="10"
  >
    <span
      className="hour__text"
    >
      10:00
    </span>
  </div>
  <div
    className="hour"
    key="11"
  >
    <span
      className="hour__text"
    >
      11:00
    </span>
  </div>
  <div
    className="hour"
    key="12"
  >
    <span
      className="hour__text"
    >
      12:00
    </span>
  </div>
  <div
    className="hour"
    key="13"
  >
    <span
      className="hour__text"
    >
      13:00
    </span>
  </div>
  <div
    className="hour"
    key="14"
  >
    <span
      className="hour__text"
    >
      14:00
    </span>
  </div>
  <div
    className="hour"
    key="15"
  >
    <span
      className="hour__text"
    >
      15:00
    </span>
  </div>
  <div
    className="hour"
    key="16"
  >
    <span
      className="hour__text"
    >
      16:00
    </span>
  </div>
  <div
    className="hour"
    key="17"
  >
    <span
      className="hour__text"
    >
      17:00
    </span>
  </div>
  <div
    className="hour"
    key="18"
  >
    <span
      className="hour__text"
    >
      18:00
    </span>
  </div>
  <div
    className="hour"
    key="19"
  >
    <span
      className="hour__text"
    >
      19:00
    </span>
  </div>
  <div
    className="hour"
    key="20"
  >
    <span
      className="hour__text"
    >
      20:00
    </span>
  </div>
  <div
    className="hour"
    key="21"
  >
    <span
      className="hour__text"
    >
      21:00
    </span>
  </div>
  <div
    className="hour"
    key="22"
  >
    <span
      className="hour__text"
    >
      22:00
    </span>
  </div>
  <div
    className="hour"
    key="23"
  >
    <span
      className="hour__text"
    >
      23:00
    </span>
  </div>
</div>
`;
